### DISCUSSION TITLE
the default save path of log

### DISCUSSION THREAD

**Discussion by jimingda** [robotology/icub-gazebo-grasping-sandbox#54] [Q&A] [answered]
Hi,

I made a test  inside docker image, replace the object of my own model with the default pudding_box on the table in `icub-gazebo-grasping-sandbox-test.sdf`, and it seems that segment failed because the `pc_object.off` and `pc_table.off` didn't change along with pc_scene.off.
```
09:24 $ icub-grasp.sh test
test is starting up...
[INFO] |yarp.os.Port|/root| Port /root active at tcp://172.17.0.2:10000/
[ERROR] |yarp.os.Property| cannot read from ymanager.ini
[INFO] |yarp.os.Network| Waiting for /icub-grasp/rpc...
Response: [fail]
...test done
```

```
-rw-r--r--  1 gitpod gitpod      11 Oct 20 09:17 pc_object.off
-rw-r--r--  1 gitpod gitpod 3519822 Oct 20 09:25 pc_scene.off
-rw-r--r--  1 gitpod gitpod 3478091 Oct 20 09:17 pc_table.off
```
My question is where can i find the log file of app, that i can debug my test.

Another question is that how can i launch the app through yarpmanager, i tried the below command after launch and connect 
the app, but don't know what's the next command, despite that i can find the command 'segment' 'fit' 'grasp' etc in `rpc_IDL.cpp `
If i can launch the app through yarpmanager, then i know how to export the log.
```
yarp wait /icub-grasp/rpc
echo "go on" | yarp rpc /icub-grasp/rpc
```

Thanks in advance.

**Comment by pattacini** [comment, marked as answer]
>My question is where can i find the log file of app, that i can debug my test.

The relevant part that I guess you're referring to is:

https://github.com/robotology/icub-gazebo-grasping-sandbox/blob/c9e5edd8c40b6af602dfdfafa41a16516ca2383a/src/main.cpp#L398-L410

>Another question is that how can i launch the app through yarpmanager

Main apps are under https://github.com/robotology/icub-gazebo-grasping-sandbox/tree/master/scripts.
The supervisor script `icub-grasp.sh` launches the XML app and interacts with it. Check it out.

> [!important]
This sandbox is meant for showcasing some basic capabilities of the robot. You can certainly experiment with it, but most likely you'll have to replace some components (e.g., segmentation, fitting...) if you aim to reach high accuracy in real scenarios.

**Reply by jimingda** [reply]
Thanks @pattacini 

In `icub-grasp.xml`, i add the following part, 
```
    <module>
        <name>yarplogger</name>
        <parameters>--start</parameters>
        <node>localhost</node>
    </module>
```
and add the env `<environment>YARP_FORWARD_LOG_ENABLE=1</environment>` in `icub-gazebo-grasping-sandbox` module.
Then the log can be exported from yarplogger.
```
    <module>
        <name>icub-gazebo-grasping-sandbox</name>
        <environment>YARP_FORWARD_LOG_ENABLE=1</environment>
        <parameters></parameters>
        <dependencies>
            <port timeout="20">/iKinGazeCtrl/rpc</port>
            <port timeout="20">/icubSim/cartesianController/right_arm/state:o</port>
            <port timeout="20">/icubSim/cartesianController/left_arm/state:o</port>
        </dependencies>
        <node>localhost</node>
    </module>
```
